src/models/context_tag_keys.rs

```diff
@@ -181,3 +182,34 @@ impl Tags {
         self.0.get(key)
     }
 }
+
+pub(crate) static TAG_KEY_LOOKUP: Lazy<BTreeMap<&'static str, ContextTagKey>> = Lazy::new(|| {
```

Comment:
@frigus02, I figure this might be a tad controversial, but it seemed like the best way to keep your length checks, etc. all intact without a huge overhaul. 😄

Reply:
Yeah, this seems fine to me. We can think about a refactor later. It keeps the length check, which is the important bit.